pkg/serverless/trace/propagation/carriers.go

```diff
@@ -203,6 +216,29 @@ func snsEntityCarrier(event events.SNSEntity) (tracer.TextMapReader, error) {
 	return carrier, nil
 }
 
+// handleEventBridgeThroughSNS builds payload carrier for an EventBridge
+// event wrapped in an SNS message.
+func handleEventBridgeThroughSNS(eventBridgeEvent map[string]interface{}) (tracer.TextMapReader, error) {
+	detail, ok := eventBridgeEvent[detailHeader].(map[string]interface{})
+	if !ok {
+		return nil, errorUnsupportedPayloadType
+	}
+
+	traceContextAny, ok := detail[datadogTraceHeader].(map[string]interface{})
+	if !ok {
+		return nil, errorNoDDContextFound
+	}
+
+	// Some values are integers, not strings, so we have to convert before using tracer.TextMapCarrier()
+	traceContext := make(map[string]string)
```

Comment:
Since we know how many keys we're going to put into this map, we can make the map and specify it's length from the get go. This will save time that would otherwise have been taken up performing allocations in order to increase the memory allocated for this map.

```go
traceContext := make(map[string]string, len(traceContextAny))
```

Reply:
Great idea!